ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1221,9 +1231,31 @@ private Map<String, LocalResource> addTempResources(Configuration conf, String h
         LOG.info("Skipping vertex resource " + file + " that already exists in the session");
         continue;
       }
-      Path hdfsFilePath = new Path(hdfsDirPathStr, getResourceBaseName(new Path(file)));
-      LocalResource localResource = localizeResource(new Path(file),
-          hdfsFilePath, type, conf);
+      Path filePath = new Path(file);
+      Path hdfsFilePath = new Path(hdfsDirPathStr, getResourceBaseName(filePath));
+      LocalResource localResource = null;
+      /*
+       * This method can be called with HDFS resources, which leads to a 'Wrong FS' IllegalArgumentException
+       * while trying to obtain a Path of a file on hdfs from a RawLocalFileSystem.
+       * A typical example is a usecase like below (sha optimization is used only for local resources):
+       * set hive.aux.jars.path=hdfs:///tmp/test_load_aux_jar/some.jar;
+       */
+      String scheme = filePath.getFileSystem(conf).getScheme();
+      if (!"file".equalsIgnoreCase(scheme)) {
```

Comment:
There is a similar duplication as part of CopyUtils#isLocal(FileSystem fs), maybe a static method can be created in FileUtils.

Reply:
let me check this one and create suitable method in FileUtils if needed